pkg/action/operator_list_operands.go

```
@@ -0,0 +1,183 @@
+	csv := v1alpha1.ClusterServiceVersion{}
+	csvKey := types.NamespacedName{}
+
+	if operator.Status.Components == nil {
+		return nil, fmt.Errorf("could not find underlying components for operator %s", operator.Name)
+	}
+	for _, resource := range operator.Status.Components.Refs {
+		if resource.Kind == v1alpha1.ClusterServiceVersionKind {
+			csvKey.Name = resource.Name
+			csvKey.Namespace = resource.Namespace
+			break
+		}
+	}
+
+	if csvKey.Name == "" && csvKey.Namespace == "" {
+		return nil, fmt.Errorf("could not find underlying CSV for operator %s", operator.Name)
+	}
+
+	err := o.config.Client.Get(ctx, csvKey, &csv)
+	if err != nil {
+		return nil, fmt.Errorf("could not get %s CSV on cluster: %s", csvKey.String(), err)
+	}
+
+	// check if owned CRDs are defined on the csv
+	if len(csv.Spec.CustomResourceDefinitions.Owned) == 0 {
+		return nil, fmt.Errorf("no owned CustomResourceDefinitions specified on CSV %s, no custom resources to display", csvKey.String())
+	}
+
+	// check CSV is not in a failed state (to ensure some OLM multitenancy rules are not violated)
+	if csv.Status.Phase != v1alpha1.CSVPhaseSucceeded {
+		return nil, fmt.Errorf("CSV underlying operator is not in a succeeded state: custom resource list may not be accurate")
+	}
+
+	return csv.Spec.CustomResourceDefinitions.Owned, nil
+}
+
+// List takes in a CRD description and finds the associated CRs on-cluster.
+// List can return a potentially unbounded list that callers may need to paginate.
+func (o *OperatorListOperands) list(ctx context.Context, crdDesc v1alpha1.CRDDescription, namespaces []string) (*unstructured.UnstructuredList, error) {
+	result := &unstructured.UnstructuredList{}
+
+	// get crd group from crd name
+	crd := apiextensionsv1.CustomResourceDefinition{}
+	crdKey := types.NamespacedName{
+		Name: crdDesc.Name,
+	}
+	err := o.config.Client.Get(ctx, crdKey, &crd)
+	if err != nil {
+		return nil, nil
```

Need to return an error here if one occurs.
```suggestion
		return nil, fmt.Errorf("get crd %q: %v", crdKey.String(), err)
```